packages/web3-providers-http/test/integration/request.test.ts

```diff
@@ -0,0 +1,33 @@
+import {
+	EthExecutionAPI,
+	Web3APIPayload,
+	JsonRpcResponse,
+	JsonRpcResponseWithResult,
+} from 'web3-common';
+import { toWei } from 'web3-utils';
+import HttpProvider from '../../src/index';
+import { accounts, clientUrl } from '../fixtures/config';
+
+describe('HttpProvider - implemented methods', () => {
+	let httpProvider: HttpProvider;
+	let jsonRpcPayload: Web3APIPayload<EthExecutionAPI, 'eth_getBalance'>;
+
+	beforeAll(() => {
+		httpProvider = new HttpProvider(clientUrl);
+		jsonRpcPayload = {
+			jsonrpc: '2.0',
+			id: 42,
+			method: 'eth_getBalance',
+			params: [accounts[0].address, 'latest'],
+		} as Web3APIPayload<EthExecutionAPI, 'eth_getBalance'>;
+	});
+
+	describe('httpProvider.request', () => {
+		it('should return expected response', async () => {
+			const response: JsonRpcResponse = await httpProvider.request(jsonRpcPayload);
+			expect(
+				String(parseInt(String((response as JsonRpcResponseWithResult).result), 16)),
```

Comment:
Should use [hexToNumber](https://github.com/ChainSafe/web3.js/blob/977a2d39f934bb0b188fd79e2ae233bf71183163/packages/web3-utils/src/converters.ts#L101) function from `web3-utils`

Reply:
done